core/src/main/java/org/apache/iceberg/DataFiles.java

```diff
@@ -108,6 +108,31 @@ public static DataFile fromManifest(ManifestFile manifest) {
         .build();
   }
 
+  public static DataFile fromPositionDelete(DeleteFile deleteFile, PartitionSpec spec) {
```

Comment:
Does this have to be specific to position deletes? Will constructing `DataFile` from an equality delete file be any different?

Reply:
Good point